wrapper/pre_copy.py

```diff
@@ -184,29 +203,91 @@ def auth_cb(cred, _):
         cred_info = [[libvirt.VIR_CRED_AUTHNAME, libvirt.VIR_CRED_PASSPHRASE],
                      auth_cb, None]
         conn = libvirt.openAuth(self._uri, cred_info)
+        domxml = conn.lookupByName(self._vm_name).XMLDesc()
+        logging.debug('Fetched domxml: \n%s', domxml)
 
-        return conn.lookupByName(self._vm_name).XMLDesc()
+        return domxml
 
     def get_disks_from_config(self, config):
         return [x for x in config.hardware.device
                 if isinstance(x, vim.vm.device.VirtualDisk)]
 
+    def get_disk_by_key(self, config, key):
+        disks = [x for x in config.hardware.device
+                 if isinstance(x, vim.vm.device.VirtualDisk) and x.key == key]
+        if len(disks) != 1:
+            raise RuntimeError('Integrity error: '
+                               'Cannot find disk with key %s' %
+                               key)
+        return disks[0]
+
+    def create_snapshot(self):
+        vm = self.get_vm()
+        logging.debug('Creating snapshot to get a new change_id')
+        WaitForTask(vm.CreateSnapshot(name='v2v_cbt',
```

Comment:
Well, any exception will just error out everything. I think it would make sense to catch those we need to immediately respond to or fix, but what would we do here?
I must admit that I do not know what the error from `WaitForTask` is, but I will figure it out because the plan is to have the warm phase be able to recover from a failed copy iteration.